### ISSUE TITLE
Getting Linker error 1120 when link against DLL on windows

### ISSUE THREAD

**Issue by dotChris90** [eProsima/Fast-CDR#109]
Hello Fast-DDS team!,

I was building your project on windows with option shared library (DLL) and everything was working fine. 
But when i try to build a consuming executable for testing purpose i get the error : 

error LNK2019: unresolved external symbol "public: static enum eprosima::fastcdr::Cdr
::Endianness const eprosima::fastcdr::Cdr::DEFAULT_ENDIAN"

You can see the corresponding CMakeLists.txt here --> https://github.com/conan-io/conan-center-index/pull/5615/files#diff-360491349ca45a6797cac6ec31e60d788bf8d5ab9ade3e57fc3cb7f4e07f5867

I just curious if the error is known or not to get a feeling where to search deeper. 

It is a little bit odd for 2 reasons 
1) according to CMakeLists.txt - it should be compiled without this because "message("${BIG_ENDIAN}")"  shows 0
2) the static library is linked perfectly fine. 

Some more detailed background : 

i would like to bring Fast-DDS to the Conan Center Index with PR on --> https://github.com/conan-io/conan-center-index/pull/5615 

Conan is a dependency / package manager for C++ with a strong focus on Cross compilation and flexibility. 

Basically conan has functionality to get source code, install dependency (libs, etc.) building your own DLL or SO etc. and package (decide what is visible to other consuming projects). 

Before getting Fast-DDS into Conan Center the corresponding other parts must get into it - Fast-CDR as example. 

thanks for your time. :)

**Comment by MiguelCompany** [comment]
@dotChris90 I didn't do a thorough investigation, but I've seen that you are using `fastcdr::fastcdr` on the `target_link_libraries` step. Could you try using just `fastcdr`?

**Comment by dotChris90** [comment]
thanks for quick reply :) 

I was adapting the CMakeLists in a way the lib and dll also getting prefix lib and so linking library is --> libfastcdr-1.0 

Then I used the environment variable CONAN_LIBS (containing my depenending libs here --> just fast-cdr). 

message("target linked --> ${CONAN_LIBS}")
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})

Leads TO :
target linked --> libfastcdr-1.0

but still same. 
It's not a super "urgent" question from company - so dont worry.

**Comment by dotChris90** [comment]
maybe steps to reproduce : 

- clone fast-cdr 
- using the Windows C++ Visual Studio Tools 2019
- build with shared libs option ON
- having a "consuming test project" with this code https://github.com/conan-io/conan-center-index/pull/5615/files#diff-8722dd4adf3828a21fcaec5376be6e62fb13b55817e28925444071840dd59731 and a CMake file with 
    
    add_executable(example test_package.cpp)
    target_link_libraries(example "${LIB_PATH}/libfastcdrd-1.0.lib")

- my question would be :
What Header file is planned to be used on Windows by a consuming project? 
i just ask because i am not 100% if all symbols are exported

**Comment by dotChris90** [comment]
@MiguelCompany  - i just saw when i take your provided cmake file everything works fine. 

I was little confused before because i though it's the library name fastcdr but it is not - it is the target name and your cmake file provide relevant information i guess for windows. 

Anyway will check again ur cmake file and adapt my conan package to it.
